cli/azd/cmd/auth_serve.go

```diff
@@ -0,0 +1,141 @@
+package cmd
+
+import (
+	"context"
+	"encoding/json"
+	"fmt"
+	"io"
+	"log"
+	"net/http"
+	"os"
+	"os/signal"
+	"syscall"
+	"time"
+
+	"github.com/Azure/azure-sdk-for-go/sdk/azcore"
+	"github.com/Azure/azure-sdk-for-go/sdk/azcore/policy"
+	"github.com/azure/azure-dev/cli/azd/cmd/actions"
+	"github.com/azure/azure-dev/cli/azd/pkg/auth"
+	"github.com/azure/azure-dev/cli/azd/pkg/input"
+	"github.com/azure/azure-dev/cli/azd/pkg/output"
+	"github.com/spf13/cobra"
+)
+
+// TokenResponse defines the structure of the token response.
+type TokenResponse struct {
+	AccessToken string `json:"access_token"`
+	ExpiresOn   int64  `json:"expires_on"`
+}
+
+// tokenHandler handles token requests.
+func (serve *serveAction) tokenHandler(w http.ResponseWriter, r *http.Request) {
+	resource := r.URL.Query().Get("resource")
```

Comment:
Do we need to ensure anything about the verb that was used (I thought that MSI uses `POST`, not `GET`, for some reason?)

Reply:
https://learn.microsoft.com/en-us/entra/identity/managed-identities-azure-resources/how-to-use-vm-token#get-a-token-using-http
it uses get

Reply:
We should ensure that the `Metadata` header is present, as outlined on the linked page and reject other verbs, then.